pkg/staticnetworkconfig/generator.go

```diff
@@ -308,6 +309,67 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 	return nil
 }
 
+func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistenceYAML(macInterfaceMap models.MacInterfaceMap, networksYaml string) error {
+	interfaceNames := lo.Map(macInterfaceMap, func(m *models.MacInterfaceMapItems0, _ int) string { return m.LogicalNicName })
+
+	var config map[string]interface{}
+
+	// Unmarshal the JSON string into the config struct
+	err := yaml.Unmarshal([]byte(networksYaml), &config)
+	if err != nil {
+		s.log.WithError(err).Errorf("Error unmarshalling JSON")
+		return err
+	}
+
+	// See 'man systemd.net-naming-scheme' for interface naming protocol
+	predicatableIfaceNamePattern := regexp.MustCompile("^en[PsvxXbucaipod]")
+	Interfaces := config["interfaces"].([]interface{})
+
+	for _, iface := range Interfaces {
+		nic := iface.(map[interface{}]interface{})
+		interfaceName := nic["name"]
+		interfaceType := nic["type"]
+		switch interfaceType {
+		case "802-3-ethernet", "ethernet":
+			if !lo.Contains(interfaceNames, interfaceName.(string)) {
+				if !predicatableIfaceNamePattern.MatchString(interfaceName.(string)) {
+					return errors.Errorf("mac-interface mapping for interface %s is missing and not a physical interface", interfaceName)
+				} else {
+					s.log.Infof("Interface %s has no mac-interface mapping but matches a physical interface", interfaceName)
+				}
+			}
+		case "bond":
+			if val, exists := nic["link-aggregation"].(map[interface{}]interface{}); exists {
+				if ports, exists := val["port"].([]interface{}); exists {
+					for _, port := range ports {
+						if !lo.Contains(interfaceNames, port.(string)) {
+							if !predicatableIfaceNamePattern.MatchString(interfaceName.(string)) {
```

Comment:
We also need to ignore the missing name when the interface name appears as an interface in the NMState itself.

Reply:
I believe that the following won't work when using nmstate.service.

```
 ---
 interfaces:
 - name: eth0
   type: ethernet
   state: up
   identifier: mac-address
   mac-address: 00:23:45:67:89:1a
 - name: eth1
   type: ethernet
   state: up
   identifier: mac-address
   mac-address: 00:23:45:67:89:1b
 - name: bond0
   type: bond
   state: up
   link-aggregation:
     mode: balance-rr
     port:
       - eth0
       - eth1
 ```

If it is indeed the case, the bond validation should remain as is.

Reply:
@AlonaKaplan  As discussed offline, the above example indeed doesn't work with nmstate, but it also fails with a VerificationError when the ports are physical interface names. I've reached out to Gris to check if this is a known issue with nmstate.

Reply:
Gris said [here](https://github.com/nmstate/nmstate/issues/2380#issuecomment-2316805780) that it would work for `nmstatectl apply`. Is there something different about `nmstatectl service` that would make it not work there? It seems to me like an issue with nmstate if this doesn't work. Forbidding this config would certainly be a regression for ABI.

Reply:
@cathay4t would the above config work using nmstate apply?

Reply:
@zaneb I’ve added a temporary condition to prevent regressions for ABI while blocking non-ABI users from using the mac-identifier - https://github.com/openshift/assisted-service/pull/6695/commits/f426f0976d656a7dde08f1693b8cc528e95b7e52#diff-8a1899b4c0132785a200ded4829d37bdd9c27e5f182f9edd91879a228649a45cR327-R329

Reply:
YAML at https://github.com/openshift/assisted-service/pull/6695#discussion_r1759656407 will not works as expected, the bond still refer the bond port by name eth1 and eth2.

You may use `interface.controller` property for now:
```yml
---
interfaces:
  - name: bond0
    type: bond
    link-aggregation:
      mode: balance-rr
  - name: bond0-port1
    type: ethernet
    state: up
    identifier: mac-address
    mac-address: 00:23:45:67:89:1a
    controller: bond0
```

Another approach would be wait https://github.com/nmstate/nmstate/pull/2710

Reply:
@zaneb as Gris approved, the [yaml](https://github.com/openshift/assisted-service/pull/6695#discussion_r1759656407) is not supported by nmstate and not the current/planned API to refer bond ports with mac-identifier.
I understand it works with nmstate generator, but it is kind of a hack.
I think both ABI and AI should be aligned with the official nmstate API and not support this format.